packages/flame/lib/src/experimental/viewfinder.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+import '../components/component.dart';
+import '../game/transform2d.dart';
+import 'camera.dart';
+import 'viewport.dart';
+
+/// [Viewfinder] is a part of a [Camera2] system that controls which part of
+/// the game world is currently visible through a viewport.
+///
+/// The viewfinder contains the game point that is currently at the
+/// "cross-hairs" of the viewport ([position]), the [zoom] level, and the
+/// [angle] of rotation of the camera.
+///
+class Viewfinder extends Component {
+  /// Internal transform matrix used by the viewfinder.
+  final Transform2D _transform = Transform2D();
+
+  /// The game coordinates of a point that is to be positioned at the center
+  /// of the viewport.
+  Vector2 get position => -_transform.position;
+  set position(Vector2 value) => _transform.position = -value;
+
+  /// Zoom level of the game.
+  ///
+  /// The default zoom value of 1 means that the world coordinates are in 1:1
+  /// correspondence with the pixels on the screen. Zoom levels higher than 1
+  /// make the world appear closer: each unit of game coordinate systems maps
+  /// to [zoom] pixels on the screen. Conversely, when [zoom] is less than 1,
+  /// the game world will appear further away and smaller in size.
+  ///
+  /// See also: [visibleGameSize] for setting the zoom level dynamically.
+  double get zoom => _transform.scale.x;
+  set zoom(double value) {
+    assert(value > 0, 'zoom level must be positive: $value');
+    _transform.scale = Vector2.all(value);
+  }
+
+  /// Rotation angle of the game world, in radians.
+  ///
+  /// The rotation is around the axis that is perpendicular to the screen.
+  double get angle => -_transform.angle;
+  set angle(double value) => _transform.angle = -value;
+
+  /// Reference to the parent camera.
+  Camera2 get camera => parent! as Camera2;
```

Comment:
what prevents users from manually adding a Viewfinder to other components? I wonder if there is a neat place to add a helpful error

Reply:
There is an assert in `onMount`, I think that should suffice.